src/amo/components/Search/SearchForm.js

```diff
@@ -0,0 +1,70 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { loadEntities } from 'core/actions';
+import { fetchAddon } from 'core/api';
+import { gettext as _ } from 'core/utils';
+
+import 'amo/css/SearchForm.scss';
+import 'amo/css/lib/buttons.scss';
+
+export class SearchFormBase extends React.Component {
+  static propTypes = {
+    pathname: PropTypes.string.isRequired,
+    query: PropTypes.string.isRequired,
+  }
+  static contextTypes = {
+    router: PropTypes.object,
+  }
+
+  goToSearch(query) {
+    const { pathname } = this.props;
+    this.context.router.push(`${pathname}?q=${query}`);
+  }
+
+  handleSearch = () => {
+    this.goToSearch(this.refs.query.value);
+  }
+
+  handleKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      this.handleSearch();
+    }
+  }
+
+  render() {
+    const { query } = this.props;
+    return (
+      <form ref="form" className="search-form" method="GET"
+            action="/en-GB/firefox/search/" onKeyDown={this.handleKeyDown}
```

Comment:
Oh, I suppose removing it and letting the `keyDown` just fire `submit` (which we handle) is better anyway and might avoid needing to do the cross-compat check on submit.